diag_manager/fms_diag_yaml.F90

```diff
@@ -84,14 +85,18 @@ module fms_diag_yaml_mod
 !> @brief type to hold the diag_file information
 type diagYamlFiles_type
   character (len=:), private, allocatable :: file_fname !< file name
-  character (len=:), private, allocatable :: file_frequnit !< the frequency unit
+  integer,           private              :: file_frequnit !< the frequency unit (DIAG_SECONDS, DIAG_MINUTES, &
+                                                           !! DIAG_HOURS, DIAG_DAYS, DIAG_YEARS)
   integer, private    :: file_freq !< the frequency of data
-  character (len=:), private, allocatable :: file_timeunit !< The unit of time
+  integer,           private              :: file_timeunit !< The unit of time (DIAG_SECONDS, DIAG_MINUTES, &
+                                                           !! DIAG_HOURS, DIAG_DAYS, DIAG_YEARS)
   character (len=:), private, allocatable :: file_unlimdim !< The name of the unlimited dimension
   type(subRegion_type), private :: file_sub_region !< type containing info about the subregion, if any
   integer, private :: file_new_file_freq !< Frequency for closing the existing file
-  character (len=:), private, allocatable :: file_new_file_freq_units !< Time units for creating a new file.
+  integer,           private              :: file_new_file_freq_units !< Time units for creating a new file.
```

Comment:
I think I missed this: where does the units get converted to an integer?

Reply:
The units are read to a string buffer and the converter to a integer.
https://github.com/NOAA-GFDL/FMS/blob/8c54912c10915a37c97603a2fcf35aeb584bfc6a/diag_manager/fms_diag_yaml.F90#L475-L478

https://github.com/NOAA-GFDL/FMS/blob/8c54912c10915a37c97603a2fcf35aeb584bfc6a/diag_manager/fms_diag_yaml.F90#L678-L679